### PR TITLE
Gallery implementation

### LINKED ISSUE #93: #92 Implement the gallery, show hard-coded images in gallery
Implement so that the gallery activity shows thumbnails of hardcoded pictures, and that the pictures are able to be pressed to expand the thumbnail. If time that you can swipe through the gallery as well.

### PR DESCRIPTION
**Pull request by aolsson711**
- Thumbnails in gallery (Hardcoded with image from resources)
- New activity for when pressing a thumbnail, to view full screen of image)

TODO comments added for changing from getting images from resource to using paths, that also will be used when taking pictures with the camera.

closes #93

### CODE REVIEW

**Review by giommok — APPROVED** [review]
Seems good.

File: app/src/main/res/layout/activity_gallery.xml [hunk 1]
```diff
@@ -3,10 +3,19 @@
...
     android:background="@drawable/background"
     tools:context=".SettingsActivity">
 
     <!-- Toolbar -->
     <include layout="@layout/toolbar"/>
 
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/gallery_recyclerview"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
```
Comment (magangloff) [inline app/src/main/res/layout/activity_gallery.xml:16]: Change layout height to display all the images at the end of the gallery.
I don't know why, but it seems like Top to bottom of the toolbar is not enough to make gallery start after toolbar, thus I added paddingTop with the height of the toolbar
```suggestion
        android:layout_height="match_parent"
        android:paddingTop="80dp"
```

File: app/src/main/res/layout/activity_gallery.xml [hunk 2]
```diff
@@ -3,10 +3,19 @@
...
+        app:layout_constraintEnd_toEndOf="parent"
```
Comment (magangloff) [inline app/src/main/res/layout/activity_gallery.xml:17]: End to end constraint can be removed
```suggestion

```

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryAdapter.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+import androidx.recyclerview.widget.RecyclerView;
+
+import java.util.List;
+
+/**
+ * Adapter for gallery to show all images
+ */
+public class GalleryAdapter extends RecyclerView.Adapter<GalleryAdapter.ViewHolder> {
+
+    private Context mContext;
```
Comment (magangloff) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryAdapter.java:19]: mContext can be made final
```suggestion
    private final Context mContext;    
```

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryAdapter.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    // TODO change to list of paths to images in folder
+    private List<Integer> images;
```
Comment (magangloff) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryAdapter.java:21]: images can be made final
```suggestion
    private final List<Integer> images;
```

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryAdapter.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    @Override
+    public int getItemCount() {
+        return images.size();
+    }
+
+    /**
+     * View holder for the gallery adapter.
+     * Contains the image view to be able to set the correct image.
+     */
+    public class ViewHolder extends RecyclerView.ViewHolder {
```
Comment (magangloff) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryAdapter.java:79]: Class Viewholder can be made static
```suggestion
    public static class ViewHolder extends RecyclerView.ViewHolder {
```

File: app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryAdapter.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        private ImageView image;
```
Comment (magangloff) [inline app/src/main/java/com/github/bgabriel998/softwaredevproject/GalleryAdapter.java:80]: image can be made final
```suggestion
        private final ImageView image;
```

**Review by magangloff — CHANGES_REQUESTED** [review]
Some few changes to be made for the gallery to be properly displayed

File: app/src/main/res/layout/activity_gallery.xml [hunk 7]
```diff
@@ -3,10 +3,19 @@
...
 
-</androidx.constraintlayout.widget.ConstraintLayout>
\ No newline at end of file
+    <androidx.recyclerview.widget.RecyclerView
+        android:id="@+id/gallery_recyclerview"
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:paddingTop="80dp"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintTop_toBottomOf="@+id/toolbar"/>
```
Comment (magangloff) [inline app/src/main/res/layout/activity_gallery.xml]: Place the toolbar after the RecyclerView to make sure that the recyclerView does not overdraw the toolbar and makes the back button not usable. This makes the test TestToolbarBackButton fail.
```suggestion
</androidx.constraintlayout.widget.ConstraintLayout> 
    <androidx.recyclerview.widget.RecyclerView
        android:id="@+id/gallery_recyclerview"
        android:layout_width="match_parent"
        android:layout_height="match_parent"
        android:paddingTop="80dp"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toBottomOf="@+id/toolbar"/>
        
        <!-- Toolbar -->
        <include layout="@layout/toolbar"/>
```

**Review by magangloff — CHANGES_REQUESTED** [review]
Place toolbar after the recyclerView to use the back button.

**Review by magangloff — APPROVED** [review]

**Pull request merged**